routes/api/search.js

```diff
@@ -1,36 +1,95 @@
 
 // @route  POST api/search
-// @desc   Add Looking For request
+// @desc   Add search request
 // @access Private
 
-router.post("/", (req, res) => {
-    res.send("Add an LF request");
+router.post("/", async ({ body }, res) => {
+    try {
+        const search = await databaseControllers.addSearch(body);
+        res.status(201).send(search);
+        return;
+    } catch (error) {
+        res.status(500).send(
+            `An error occured while trying to create your search, ${error}`
+        );
+        return;
+    }
 });
 
 // @route  DELETE api/search/:id
-// @desc   Delete your LF Request
+// @desc   Delete search request
 // @access Private
 
-router.delete("/:id", (req, res) => {
-    res.send("Delete your own LF request");
+router.delete("/:searchID", async ({ params }, res) => {
+    try {
+        const search = await databaseControllers.deleteSearch(params);
+        res.status(200).send(search);
+        return;
+    } catch (error) {
+        if (error.message === "NOT_FOUND") {
+            res.status(404).send(`Search ID does not exist`);
+            return;
+        } else {
+            res.status(500).send(
+                `An error occured while trying to delete your search, ${error}`
+            );
+            return;
+        }
+    }
 });
 
 // @route  POST api/search/:id/join/:playerID
-// @desc   Manually add someone to your request
+// @desc   Join search request
 // @access Private
 
-router.post("/:id/player/:playerID", (req, res) => {
-    res.send("Add a player to an LF Request");
+router.post("/:searchID/player/:playerID", async ({ params }, res) => {
+    try {
+        const search = await databaseControllers.joinSearch(params);
+        res.status(201).send(search);
+        return;
+    } catch (error) {
+        if (error.message === "NOT_FOUND") {
+            res.status(404).send(`Search ID does not exist`);
+            return;
+        } else if (error.message === "ALREADY_EXISTS") {
+            res.status(409).send(`Player ID is already in Search`);
+            return;
+        } else {
+            res.status(500).send(
+                `An error occured while trying to join the search`
+            );
+            return;
+        }
+    }
 });
 
 // @route  DELETE api/search/:id/join/::playerID
-// @desc   Join/Unjoin an LF request
+// @desc   Unjoin search request
 // @access Private
 
-router.delete("/:id/player/:playerID", (req, res) => {
-    res.send("Remove a player from an LF Request");
+router.delete("/:searchID/player/:playerID", async ({ params }, res) => {
+    try {
+        const search = await databaseControllers.leaveSearch(params);
+        res.send(search);
+        return;
+    } catch (error) {
+        if (error.message === "NOT_FOUND") {
+            res.status(404).send(`Search ID does not exist`);
+            return;
+        } else if (error.message === "PLAYER_NOT_FOUND") {
+            res.status(404).send(`Player ID is not in Search`);

```

Comment:
Did you intend to not add the error message for this unexpected generic error? (e.g. `error` or `error.message`)?